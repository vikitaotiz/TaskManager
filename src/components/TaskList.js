import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import fetchTasksFromApi from '../redux/apiCalls';
import { fetchTasks } from '../redux/tasks/tasks';

const TaskList = () => {
  const tasks = useSelector((state) => state.tasks);
  const dispatch = useDispatch();

  useEffect(async () => {
    const data = await fetchTasksFromApi();
    dispatch(fetchTasks(data));
  }, []);

  return (
    <div className="container mt-3">
      <ul className="list-group">
        {tasks.length > 0 ? tasks.map((task) => (
          <li key={task.id} className="list-group-item mb-2 rounded">
            {task.title}
          </li>
        ))
          : 'No Data'}
      </ul>
    </div>
  );
};

export default TaskList;
