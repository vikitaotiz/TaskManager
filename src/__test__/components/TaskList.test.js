import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import TaskList from '../../components/TaskList';

it('should render correctly', () => {
  const body = document.createElement('tbody');
  const data = [
    { id: 1, title: 'Task One', userId: 1 },
    { id: 2, title: 'Task Two', userId: 1 },
  ];

  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <TaskList data={data} />
        </BrowserRouter>
      </Provider>,
      body,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
