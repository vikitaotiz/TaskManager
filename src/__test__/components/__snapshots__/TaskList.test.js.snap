// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container mt-3"
>
  <div
    className="mb-4 d-flex justify-content-between"
  >
    <a
      aria-current="page"
      className="text-decoration-none text-white active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="fas fa-arrow-left"
      />
       
      Back
    </a>
    <b
      className="text-decoration-none text-white"
    >
      All Tasks
    </b>
    <span>
      <div>
        <button
          className="btn btn-primary btn-dm"
          data-bs-target="#exampleModal"
          data-bs-toggle="modal"
          type="button"
        >
          <i
            className="fas fa-plus-circle"
          />
        </button>
        <div
          aria-hidden="true"
          aria-labelledby="exampleModalLabel"
          className="modal fade"
          id="exampleModal"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h5
                  className="modal-title"
                  id="exampleModalLabel"
                >
                  Add Task
                </h5>
                <button
                  aria-label="Close"
                  className="btn-close"
                  data-bs-dismiss="modal"
                  type="button"
                />
              </div>
              <div
                className="modal-body"
              >
                
                <form
                  className="p-2 m-2"
                >
                  <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Task Title"
                    type="text"
                    value=""
                  />
                  <div
                    className="d-grid"
                  >
                    
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>
  <hr />
  <ul
    className="list-group"
  >
    No Data
  </ul>
</div>
`;
